Term addition for .std() in NumPy (#4490)
Codecademy/docs: content/numpy/concepts/built-in-functions/terms/std/std.md

---
Title: '.std()'

> Description: 'Calculates the standard deviation of given data along a specified axis'

Reviewer: ```suggestion
Description: 'Calculates the standard deviation of given data along a specified axis.'
```

Subjects:
  - 'Computer Science'
  - 'Data Science'
Tags:
  - 'Arrays'
  - 'Functions'
  - 'NumPy'
CatalogContent:
  - 'learn-python-3'
  - 'paths/data-science'
---

> The **`.std()`** function returns the standard deviation of the array elements. The standard deviation is calculated for the flattened array by default, otherwise over the specified axis.

Reviewer: You need to reword this part.

It shows up as directly plagiarized from https://www.javatpoint.com/numpy-std and https://numpy.org/devdocs/reference/generated/numpy.nanstd.html

  - **axis = 1**: Calculates the standard deviation along the horizontal axis.

  - **tuple of ints**: Calculates the standard deviation along multiple specified axes.

- `dtype`: Type used in computing the standard deviation, if specified. By default, for arrays of integer type, it is float64, while for arrays of float types, it matches the array type.

  > **Note** For floating-point inputs, the standard deviation is calculated with the same precision as the input data. This may cause inaccuracies, especially with `np.float32` data type.

- `out`: Specifies an alternative output array to contain the result. This array must have the same shape as the expected output.

- `ddof`: It stands for _Delta Degrees of Freedom_. It helps adjust the calculation of standard deviation for samples.

- `keepdims`: It accepts a boolean value and is used to determine whether to retain the dimensions of the input array in the output. By default, it is set to `False`.

- `where`: It accepts boolean arrays or conditions where `True` values indicate the indices, or elements within the array for which the standard deviation should be calculated.

> **Returns:** If `out` is `None`, it returns a new array containing the standard deviation. Otherwise, it assigns the result to the specified output array and returns its reference.

Reviewer: ```suggestion
If the `out` parameter is `None`, the `.std()` function returns a new array containing the standard deviation. Otherwise, it assigns the result to the specified output array and returns its reference.
```

---
Title: '.std()'
Description: 'Calculates the standard deviation of given data along a specified axis'
Subjects:
  - 'Computer Science'
  - 'Data Science'
Tags:
  - 'Arrays'
  - 'Functions'
  - 'NumPy'
CatalogContent:
  - 'learn-python-3'
  - 'paths/data-science'
---

The **`.std()`** function returns the standard deviation of the array elements. The standard deviation is calculated for the flattened array by default, otherwise over the specified axis.

A standard deviation is a statistical measure indicating the spread of a distribution of data, represented by an array, along a specified axis.

## Syntax

```pseudo
numpy.std(a, axis, dtype, out, ddof, keepdims, where)
```

- `a`: Array of elements used to find the standard deviation.

> ### Optional Parameters:

Reviewer 2: ```suggestion
### Optional Parameters
```

  > **Note** For floating-point inputs, the standard deviation is calculated with the same precision as the input data. This may cause inaccuracies, especially with `np.float32` data type.

- `out`: Specifies an alternative output array to contain the result. This array must have the same shape as the expected output.

- `ddof`: It stands for _Delta Degrees of Freedom_. It helps adjust the calculation of standard deviation for samples.

- `keepdims`: It accepts a boolean value and is used to determine whether to retain the dimensions of the input array in the output. By default, it is set to `False`.

- `where`: It accepts boolean arrays or conditions where `True` values indicate the indices, or elements within the array for which the standard deviation should be calculated.

**Returns:** If `out` is `None`, it returns a new array containing the standard deviation. Otherwise, it assigns the result to the specified output array and returns its reference.

> **Note** For complex numbers, `std` takes the absolute value before squaring for a real, nonnegative result.

## Example

The following examples demonstrate the use of `.std()` with different parameters.

> 
> **Example**

Reviewer 2: ```suggestion
```
